packages/0x.js/src/order_watcher/order_state_watcher.ts

```diff
@@ -123,6 +132,18 @@ export class OrderStateWatcher {
         this._orderFilledCancelledLazyStore.deleteAll();
         delete this._callbackIfExistsAsync;
         this._eventWatcher.unsubscribe();
+        this._expirationWatcher.unsubscribe();
+    }
+    private _onOrderExpired(orderHash: string): void {
+        const orderState: OrderState = {
+            isValid: false,
+            orderHash,
+            error: ExchangeContractErrs.OrderFillExpired,
+        };
+        if (!_.isUndefined(this._orderByOrderHash[orderHash])) {
+            // We need this check because we never remove the orders from expiration watcher
+            (this._callbackIfExistsAsync as OnOrderStateChangeCallback)(orderState);
```

Comment:
We should probably check if `_callbackIfExistsAsync` is defined. Then we don't need to cast it.

Reply:
@LogvinovLeon please address this feedback.